### PR TITLE
pkg/oonf_api: export strcasecmp [-Werror=implicit-function-declaration]

### PR DESCRIPTION
**Pull request by cgundogan**
Building `tests/nhdp` for a target other than `native` and with a newer compiler version fails with
```
RIOT/tests/nhdp/bin/pkg/samr21-xpro/oonf_api/src-api/common/avl_comp.c: In function 'avl_comp_strcasecmp':
RIOT/tests/nhdp/bin/pkg/samr21-xpro/oonf_api/src-api/common/avl_comp.c:146:10: error: implicit declaration of function 'strcasecmp' [-Werror=implicit-function-declaration]
   return strcasecmp(txt1, txt2);
          ^~~~~~~~~~
cc1: all warnings being treated as errors
```
`man strcasecmp` tells me that this function is located in `strings.h`, but this file is already included in `oonf_api`. I figured, however, that an export in `common/string.h` is necessary and added a patch for that.

BTW: I noticed this while testing with [Jenkins](https://riot-ci.inet.haw-hamburg.de/job/Build_PR_Multi/view/change-requests/job/PR-6258/1/artifact/), which has a slightly newer compiler than murdock. @kaspar030 what do you think about updating murdock's toolchains?

### CODE REVIEW

**Comment by kaspar030** [comment]
> man strcasecmp tells me that this function is located in strings.h, but this file is already included in oonf_api.

strcasecmp is also defined by the newlib strings.h, but we override that in the posix module. I came across the same when testing builds, my solution was to drop sys/posix/include/string.h.

@miri64 do we need that file?

**Comment by kaspar030** [comment]
> BTW: I noticed this while testing with Jenkins, which has a slightly newer compiler than murdock. 
> @kaspar030 what do you think about updating murdock's toolchains?

Hm, maybe... Thing is, we'd be either testing the old gcc *or* the newer ones. Both seems useful. :)
Are there updated PPAs?

**Comment by smlng** [comment]
@kaspar030: yes if you use the riot docker file you might simply run `apt-get update && apt-get upgrade` arm-none-eabi-gcc latest version in 5.4 in PPA.

@cgundogan: could you rebase #6250 to depend on this one, that way we should see jenkins to succeed with its tests?

**Comment by smlng** [comment]
@cgundogan, correct PR is #6258

**Comment by smlng** [comment]
ah and btw. PR fixes the described issue for gcc > 5.2, so ACK

**Comment by smlng** [comment]
> what about the header clash?

@kaspar030 you mean between newlib and posix in RIOT:

>strcasecmp is also defined by the newlib strings.h, but we override that in the posix module. I >came across the same when testing builds, my solution was to drop sys/posix/include/string.h.

IMO this should be (re)solved in a separate PR, this PR fixes the described issue with the oonf_api pkg when compiling `tests/nhdp` - and is needed ASAP to get tests running on newer gcc versions. The header clash you described seems to be not an issue here.

**Comment by kaspar030** [comment]
> The header clash you described seems to be not an issue here.

Guess again:

```
[kaspar@booze riot (master)]$ BOARD=samr21-xpro make -C tests/nhdp clean all &> /dev/null && echo OK! || echo FAIL.
FAIL.
[kaspar@booze riot (master)]$ rm sys/posix/include/strings.h
[kaspar@booze riot (master)]$ BOARD=samr21-xpro make -C tests/nhdp clean all &> /dev/null && echo OK! || echo FAIL.
OK!
[kaspar@booze riot (master)]$
```

**Comment by smlng** [comment]
@kaspar030 mhm strange, or better let me rephrase: deleting `sys/posix/include/strings.h` solves the describes issue, too. No need to patch oonf_api, so this PR might be obsolete. However, can we safely remove the posix strings.h? If so, I'd rather take that as a solution then this one.

**Comment by kaspar030** [comment]
> However, can we safely remove the posix strings.h?

@miri64 You're the author. newlib has the file, and all our apps build fine without the sys/posix version.

It does hard-define ffs() to the builtin, but the compiler should do that anyways, right?

**Comment by miri64** [comment]
> @miri64 You're the author. newlib has the file, and all our apps build fine without the sys/posix version.
> 
> It does hard-define ffs() to the builtin, but the compiler should do that anyways, right?

@kaspar030 if I remember correctly this was done for msp430 and AVR. There is no newlib (yet) on those platforms

**Comment by BytesGalore** [comment]
I ACK this PR since it fixes the actual problem with the oonf API.
It shouldn't conflict if `strings.h` (#6263) would be removed some day.

**Comment by OlegHahm** [comment]
@kaspar030, what's your assessment?

**Comment by kaspar030** [comment]
this should not be needed amymore?

**Comment by OlegHahm** [comment]
Yeah, I have the same impression. Sorry, was too lazy to read through the discussion last Friday. @cgundogan, okay with closing?

**Comment by cgundogan** [comment]
>@cgundogan, okay with closing?

sure
